client/app/scripts/superdesk-desks/desks.js

```diff
@@ -119,6 +127,163 @@
                         queryItems();
                     }
                 });
+
+                var container = elem[0];
+                var offsetY = 0;
+                elem.bind('scroll', function() {
+                    scope.$apply(function() {
+                        if (container.scrollTop + container.offsetHeight >= container.scrollHeight - 3) {
+                            container.scrollTop = container.scrollTop - 3;
+                            scope.fetchNext();
+                        }
+                        if (container.scrollTop <= 2) {
+                            offsetY = 2 - container.scrollTop;
+                            container.scrollTop = container.scrollTop + offsetY;
+                            scope.fetchPrevious();
+                        }
+                    });
+                });
+                scope.fetchNext = function() {
+                    if (!scope.fetching) {
+                        if (scope.cacheNextItems.length > 0) {
+                            scope.fetching = true;
+                            scope.page = scope.page + 1;
+
+                            criteria.source.from = (scope.page) * criteria.source.size;
+                            scope.loading = true;
+
+                            if (scope.items.length > criteria.source.size){
+                                scope.cachePreviousItems = _.slice(scope.items, 0, criteria.source.size);
+                                scope.items.splice(0, criteria.source.size);
+                            }
+                            $timeout(function() {
+                                if (!_.isEqual(scope.items, scope.cacheNextItems)) {
+                                    scope.items = scope.items.concat(scope.cacheNextItems);
+                                }
+                            }, 100);
+
+                            api('archive').query(criteria)
+                            .then(function(items) {
+                                scope.cacheNextItems = items._items;
+                                scope.fetching = false;
+                            }, function() {
```

Comment:
i think that those empty error handling functions should be removed before merge. There are few places like this in this file.